From 5c1149c3bf74aca1d2c297cf1c488870f3e8425b Mon Sep 17 00:00:00 2001
From: Matthieu Mauger <matthieux.mauger@intel.com>
Date: Wed, 31 Jul 2013 17:27:23 +0200
Subject: [PORT FROM MAIN] wifi/hotspot: update connected devices list upon leaving STA
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

BZ: 123465

When WiFi Hotspot is used without security (open), the STA connect and
disconnect events were not reported by hostapd supplicant to the framework
leading to update issues in the hotspot UI.

This patch declares a station as “authorized” as soon as the association process
is completed when no security is set. When security is enabled, the “authorized”
flag is already managed at the end of WPA authentication.

When a station is marked as “authorized”, events related to this station are
then reported to the framework.

Change-Id: Ic73dd71cb72ebdfb1db298d19c4776dd4f223d7c
Orig-Change-Id: I212c5e9c34c2af9912563e51d20b69f20c8cf152
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: yes
Signed-off-by: Matthieu Mauger <matthieux.mauger@intel.com>
---
 src/ap/hostapd.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index fd1ca2b..fed1f92 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -1453,6 +1453,7 @@ void hostapd_new_assoc_sta(struct hostapd_data *hapd, struct sta_info *sta,
 	if (!hapd->conf->ieee802_1x && !hapd->conf->wpa) {
 		os_get_time(&sta->connected_time);
 		accounting_sta_start(hapd, sta);
+		ap_sta_set_authorized(hapd, sta, 1);
 	}
 
 	/* Start IEEE 802.1X authentication process for new stations */
-- 
1.7.4.1

